data_handler/handlers/liquidable_debt/debt_handlers.py

```diff
@@ -49,45 +56,177 @@ def prepare_data(self, protocol_name: str, path: str = LOCAL_STORAGE_PATH) -> di
+                    collateral_token_index = 0
+                    collateral_keys = parsed_data[token_info[ROW_ID_FIELD_NAME]][
+                        COLLATERAL_FIELD_NAME
+                    ].keys()
+
+                    if pair[collateral_token_index] not in collateral_keys:
+                        if result.get(debt_token) and debt_token not in self.token_pairs.keys():
+                            del result[debt_token]
+                            break
+
+                        continue
+
+                    self.token_pairs.update({debt_token: pair[collateral_token_index]})
+
+        return result
+
+    def _calculate_liquidable_debt(
+            self, data: dict[int, dict[str, str | Decimal | dict[str, Decimal]]] = None,
+            max_token_prices: dict[str, dict[str, Decimal | int]] = None
+    ) -> dict:
         """
         Calculates liquidable debt based on data provided and updates an existing data.
-        :param data: Data to calculate liquidable debt for.
+        Data to calculate liquidable debt for:
+        Args: data( dict[
+            int - Row number,
+            dict[
+                str - Field name,
+                str - Token value
+                | Decimal - Field value (the price as example)
+                | dict[
+                    str - Token name,
+                    Decimal - Token value
+                ]
+            ]
+        ])
+        :Args: max_token_prices( dict[
+            str - Token name,
+            dict[
+                str - Field name,
+                Decimal - Token price
+                | int - row id
+                ]
+            ])
         :return: A dictionary of the ready liquidable debt data.
         """
-        result_data = deepcopy(data)
+        result_data = dict()
+        prices = Prices()
+        asyncio.run(prices.get_lp_token_prices())
 
-        for row_number in data:
-            state = self.state_class(verbose_user=data[row_number][USER_FIELD_NAME])
+        for debt_token, token_info in max_token_prices.items():
+            # even if it isn't per-user data, we need to provide a user ID
+            # so like that we're able to provide debt and collateral values
+            user_wallet_id = data[token_info[ROW_ID_FIELD_NAME]][USER_FIELD_NAME]
+            if not self.token_pairs.get(debt_token):
+                continue
+            collateral_token_symbol = self.token_pairs[debt_token]
+            state = self.state_class(verbose_user=user_wallet_id)
 
-            state.loan_entities[data[row_number][USER_FIELD_NAME]].debt.values = {
-                key: value
-                for key, value in data[row_number][DEBT_FIELD_NAME].items()
-            }
-            state.loan_entities[data[row_number][USER_FIELD_NAME]].collateral.values = {
-                key: value
-                for key, value in data[row_number][COLLATERAL_FIELD_NAME].items()
-            }
+            if not isinstance(state, HashstackV0State) \
```

Comment:
Make code shorted when it's possible and not affect much on readability.
`if not isinstance(state, (HashstackV0State, HashstackV1State)):`

Reply:
Done